src/effects/RealtimeEffectState.h

```diff
@@ -94,11 +109,11 @@ class RealtimeEffectState : public XMLTagHandler
    //! Stateful instance made by the plug-in
    std::shared_ptr<EffectInstance> mInstance;
 
-   struct Access;
-   struct AccessState;
-   std::shared_ptr<AccessState> mpAccessState; // Destroy before mSettings
-   std::weak_ptr<EffectSettingsAccess> mwAccess;
-
+   // This must not be reset to nullptr while a worker thread is running.
+   // In fact it is never yet reset to nullptr, before destruction.
+   // Destroy before mSettings:
+   AtomicUniquePointer<AccessState> mpAccessState{ nullptr };
```

Comment:
I think `shared_ptr` is the only way to get some guarantees about the object lifetime in MT environments.  Unique-style pointer here isn't really much different in the sense of the tread safety here than the regular unique_ptr,

Reply:
The point of this class was atomicity and RAII.

Atomicity becauase std::atomic alone has a trivial destructor.  I want a non-trivial one, which assumes (as is proper with any object) other synchronization preventing a race between destruction and other use.

I want atomicity in the assignment of the pointer after construction.  The relase memory ordering guarantees that the 
other thread using acquire won't see the pointer value before it can correctly fetch the contents from that pointer.